### PR TITLE
fix(Jenkinsfile): better way to determine actual PR commit

### PR DESCRIPTION
**Pull request by vdice**
~~[WIP]: most of Jenkinsfile commented out to debug the `Git Info` stage...~~

Ref https://github.com/deis/controller-sdk-go/issues/45

### CODE REVIEW

**Comment by vdice** [comment]
I think this may be the "best" way to derive the actual PR commit via `git` incantations aside from bringing in the Pull Request Plugin.  (Would like not to resort to the latter option unless absolutely necessary, based on experience with it in deis/jenkins-jobs)

**Comment by Joshua-Anderson** [comment]
Hah. This is quite the git black magic, but if it works LGTM. 😄

File: Jenkinsfile [hunk 1]
```diff
@@ -66,9 +66,18 @@ node('linux') {
 		go_repo = readFile('tmp/GO_LIST').trim()
 
 		if (git_branch != "remotes/origin/master") {
-			// HACK: get actual PR commit (https://github.com/deis/controller-sdk-go/issues/45)
-			sh 'git rev-parse HEAD | git log --pretty=%P -n 1 | cut -c1-40 > tmp/ACTUAL_COMMIT'
-			git_commit = readFile('tmp/ACTUAL_COMMIT').trim()
+			// Determine actual PR commit, if necessary
```
Comment (c9s) [inline Jenkinsfile:69]: You can use `env.CHANGE_ID` to detect if it's a Pull Request job.

File: Jenkinsfile [hunk 2]
```diff
@@ -66,9 +66,18 @@ node('linux') {
...
+			sh 'git rev-parse HEAD | git log --pretty=%P -n 1 --date-order > tmp/MERGE_COMMIT_PARENTS'
```
Comment (c9s) [inline Jenkinsfile:70]: You can use `git rev-parse HEAD^1` to get the first parent of the commit.

**Pull request merged**